### PR TITLE
Add basic FastAPI setup

### PR DESCRIPTION
**Pull request by joshua-s**

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+### Setup
+Install development dependencies:
+- [Docker](https://docs.docker.com/docker-for-mac/install/)
+- [Poetry](https://python-poetry.org/docs/#installation)
+
+### Running the app
+To run the API on its own, run:
+```
+docker-compose run api
+```
```
Comment (ladyshaitan) [inline api/README.md]: 💯

File: api/app/app.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+from fastapi import FastAPI
+from fastapi_sqlalchemy import DBSessionMiddleware
+
+# Import app config
+config_module = import_module(os.getenv('APP_CONFIG'))
+config = config_module.Config()
+
+# Create app
+app = FastAPI()
```
Comment (ladyshaitan) [inline api/app/app.py]: 🎉

File: docker-compose.yml [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+  postgres:
+    image: postgres:alpine
+
+  api:
+    <<: *api-environment
+    restart: always
+    image: api
+    ports:
+      - "8000:80"
+    command: /start-reload.sh
```
Comment (ladyshaitan) [inline docker-compose.yml]: Thanks for moving this into the root directory!

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "python.pythonPath": "api/.venv/bin/python"
+}
```
[the remark this answers is not in the document]
Comment (joshua-s) [reply .vscode/settings.json]: Nope! This is a vscode config that can be committed.

File: api/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+## Usage
+### Setup
+Install development dependencies:
+- [Docker](https://docs.docker.com/docker-for-mac/install/)
+- [Poetry](https://python-poetry.org/docs/#installation)
+
+### Running the app
+From within this directory, run:
+```
+docker-compose build && docker-compose up
```
[the remark this answers is not in the document]
Comment (joshua-s) [reply api/README.md]: Will using this strategy still reflect local changes to dependencies and code?

[the remark this answers is not in the document]
Comment (ladyshaitan) [reply api/README.md]: So the docker image really just looks at updates to the container environments, if they're writing CRUD routes, queries etc. that is completely separate from the docker image.  If they're adding another container or adjusting environmental variables they'll likely have to build a new docker image in that case so everyone gets the update.

In most cases, updating the docker image will be unnecessary.

File: README.md [hunk 6]
```diff
@@ -1,7 +1,9 @@
...
 ## Getting Started
-1. [Download and install](https://docs.docker.com/docker-for-mac/install/) docker on your machine.
-2. Open up your terminal.
-3. `caedus-covid> docker-compose up`
\ No newline at end of file
+### API
+See [the api documentation](api/README.md).
+
+### Frontend
+To be done.
```
[the remark this answers is not in the document]
Comment (joshua-s) [reply README.md]: Fixed!

File: api/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (joshua-s) [reply api/README.md]: Fixed!

File: README.md [hunk 7]
```diff
@@ -2,10 +2,21 @@
...
+To run both the api and frontend, run the following command in your shell of choice:
+```
+docker-compose up
+```
+
+### Rebuilding the project
+If you have added new dependencies and need to rebuild the docker image, run:
 ```
-docker-compose build && docker-compose up
+docker-compose build
```
Comment (ladyshaitan) [inline README.md]: 💯

**Pull request merged**